backtesting/_plotting.py

```diff
@@ -537,10 +537,20 @@ def __eq__(self, other):
             colors = value._opts['color']
             colors = colors and cycle(_as_list(colors)) or (
                 cycle([next(ohlc_colors)]) if is_overlay else colorgen())
-            legend_label = LegendStr(value.name)
-            for j, arr in enumerate(value, 1):
+
+            tooltip_label = value.name if isinstance(value.name, str) else ", ".join(value.name)
+
+            if isinstance(value.name, str) and len(value) > 1:
+                legend_labels = [
+                    LegendStr(f"{name}[{index}]")
+                    for index, name in enumerate(repeat(value.name, len(value)))
+                ]
+            else:
+                legend_labels = [LegendStr(item) for item in _as_list(value.name)]
```

Comment:
> the first use case will be broken, as the label will be `name[0]`

Absolutely, you're right. I didn't follow it through correctly. 😅 With no such consolidations applicable, I'd therefore prefer it simply top-down as we reason about it.

```suggestion
            if isinstance(value.name, str):
                tooltip_label = value.name
                if len(value) == 1:
                    legend_labels = [LegendStr(value.name)]
                else:
                    legend_labels = [
                        LegendStr(f"{value.name}[{i}]")
                        for i in enumerate(len(value))
                    ]
            else:
                tooltip_label = ", ".join(value.name)
                legend_labels = [LegendStr(item) for item in value.name]
```